wgpu-hal/src/metal/device.rs

```diff
@@ -117,7 +117,7 @@ impl super::Device {
         let ep_resources = &layout.per_stage_map[naga_stage];
 
         let bounds_check_policy = if stage.module.runtime_checks {
-            naga::proc::BoundsCheckPolicy::ReadZeroSkipWrite
+            naga::proc::BoundsCheckPolicy::Restrict
```

Comment:
question: I'm minorly confused by this change, though I think it's correct. Did we have an issue with MacOS, too?

Reply:
The other backends all use restrict and so the new test was failing since it's testing that the write to `container[len]` wrote to `container[len - 1]` (same for reads), read zero skip write is also more costly in terms of perf.